src/main/scala/de/johoop/jacoco4sbt/JacocoPlugin.scala

```diff
@@ -78,13 +77,22 @@ object JacocoPlugin extends Plugin {
     PathFinder(classes) ** new FileFilter {
       def accept(f: File) = IO.relativize(classes, f) match {
         case Some(file) if ! f.isDirectory && file.endsWith(".class") =>
-          val name = Locate.toClassName(file)
+          val name = toClassName(file)
           inclFilters.exists(_ accept name) && ! exclFilters.exists(_ accept name)
         case _ => false
       }
     } get
   }
 
+  private def toClassName(entry: String): String =
+    entry.stripSuffix(ClassExt).replace('/', '.')
+
+  private val ClassExt = ".class"
+
+  object autoImport {
+    val jacoco = JacocoPlugin.jacoco

```

Comment:
Whatever that is called `jacoco` and `itJacoco` probably should split into two different auto plugins, and use `def projectSettings` to inject the settings.
See http://www.scala-sbt.org/1.0/docs/Plugins.html

Reply:
I agree, this is definitely something I want to restructure to make it more modern & consistent with current plugin best practices.

As it would be a breaking change to the way that the plugin works and involve much larger changes I propose doing this (and other behavioural changes) in a 3.0 release. This way we can quickly ship 2.4.0 with SBT 1.0 support so people can upgrade and spend more time getting things right.